bin/projects.py

```diff
@@ -139,28 +139,38 @@ def fetch_icon(icon_name: str) -> None:
         f.write(svgElement.toxml())
 
 
-def path_for_icon(icon_name: str) -> Path:
-    return Path(".") / "docs" / "data" / "readme_icons" / f"{icon_name}.svg"
+def path_for_icon(icon_name: str, relative_to: Path | None = None) -> Path:
+    if relative_to is None:
+        from_path = Path.cwd()
+    else:
+        from_path = relative_to.parent
+
+    absolute_path = PROJECT_ROOT / "docs" / "data" / "readme_icons" / f"{icon_name}.svg"
 
+    return absolute_path.resolve().relative_to(from_path.resolve())
 
-def str_projects(
+
+def get_projects(
     config: list[dict[str, Any]],
     *,
     online: bool = True,
     auth: str | None = None,
-) -> str:
-    io = StringIO()
-    print = functools.partial(builtins.print, file=io)
-
+) -> list[Project]:
     if online:
         for icon in ICONS:
             fetch_icon(icon)
 
     github = Github(auth) if online else None
 
-    projects = sorted((Project(item, github) for item in config), reverse=online)
+    return sorted((Project(item, github) for item in config), reverse=online)
+
+
+def render_projects(projects: list[Project], *, include_info: bool = True, dest_path: Path):
```

Comment:
```suggestion
def render_projects(projects: list[Project], *, dest_path: Path, include_info: bool = True):
```

Optional arguments last reads better, IMO.